internal/cmd/commands/connect/connect.go

```diff
@@ -588,22 +593,38 @@ func (c *Command) Run(args []string) (retCode int) {
 	}
 
 	termInfo := TerminationInfo{Reason: "Unknown"}
+	sendSessionCancel := false
 	select {
 	case <-c.Context.Done():
 		termInfo.Reason = "Received shutdown signal"
+		sendSessionCancel = true
 	case <-timer.C:
 		termInfo.Reason = "Session has expired"
 	default:
 		if c.execCmdReturnValue != nil {
 			// Don't print out in this case, so ensure we clear it
 			termInfo.Reason = ""
+			sendSessionCancel = true
 		} else {
 			if c.connectionsLeft.Load() == 0 {
 				termInfo.Reason = "No connections left in session"
 			}
 		}
 	}
 
+	if sendSessionCancel {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
minor nit: could use a `const` to define the timeout

Reply:
Good point, updated